### PR TITLE
Clean up AppliedProjector types.

### PR DESCRIPTION
**Pull request by jtlan**
These were super confusing (particularly with `Accessor` and `Projector` already running around), so I'm clearing them out.

### CODE REVIEW

File: src/animators/animator.ts [hunk 1]
```diff
@@ -7,13 +7,15 @@ module Plottable {
...
+     * @param {{ [attr: string]: (datum: any, index: number) => any; })} attrToAppliedProjector
+     *     An object that maps attributes to functions that generate attribute values.
+     * @return {any} Animators should return the Selection or
+     *     Transition object so that plots may chain the transitions between
+     *     Animators.
      */
-    animate(selection: d3.Selection<any>, attrToAppliedProjector: AttributeToAppliedProjector): d3.Selection<any> | d3.Transition<any>;
+    animate(selection: d3.Selection<any>,
+            attrToAppliedProjector: { [attr: string]: (datum: any, index: number) => any; }
+           ): d3.Selection<any> | d3.Transition<any>;
```
Comment (aicioara) [inline src/animators/animator.ts]: Is this line needed?

Comment (aicioara) [inline src/animators/animator.ts]: (the explicit return type)

Comment (bluong) [inline src/animators/animator.ts]: Which line?  The output types?

**Comment by aicioara** [comment]
I personally don't think this is clearer. I would have preferred the original version

**Comment by jtlan** [comment]
We then have to explain what an `AppliedProjector` is though...

**Pull request closed without merging**